provisionerd/provisionerd.go

```diff
@@ -239,7 +242,16 @@ connectLoop:
 			p.opts.Logger.Warn(p.closeContext, "coderd client failed to dial", slog.Error(err))
 			continue
 		}
-		p.opts.Logger.Info(p.closeContext, "successfully connected to coderd")
+		// This log is useful to verify that an external provisioner daemon is
+		// successfully connecting to coderd. It doesn't add much value if the
+		// daemon is built-in, so we only log it on the info level if extraLogs
+		// is true. This log message is mentioned in the docs:
+		// https://github.com/coder/coder/blob/5bd86cb1c06561d1d3e90ce689da220467e525c0/docs/admin/provisioners.md#L346
+		if p.extraLogs {
```

Comment:
I'm not super excited about changing the behavior between internal and external, but could we at least rename the option to `inMemory bool`? I feel it's very specific to this case so the name doesn't quite reflect what's going on.

Reply:
If there's some way to know if the provisioner is 'built-in' we should ideally use that instead.

Reply:
AFAIK a provisioner is external if it's started with `enterprise/cli/provisionerdaemonstart.go`. I now added an "IsExternalProvisioner" flag to the provisioner's init parameters which is now set in that file.